Lambdas/GHHooks/EventHandler/Handlers/PRHandler.swift

```diff
@@ -273,4 +267,66 @@ private extension PRHandler {
+     - The user who authored the PR should be credited unless they are a code owner. Such a credit should be prominent and - as the GitHub changelog generator does - include a notation if it's that user's first merged PR.
+     - Any users who reviewed the PR (even if they requested changes or did a comments-only review without approving) should also be credited unless they are code owners. Such a credit should be less prominent than the author credit, something like a "thanks to ... for helping to review this release"
+     - The release author (user who merged the PR) should always be credited in a release, even if they're a code owner. This credit should be the least prominent, maybe even just a footnote (since it will pretty much always be a owner/maintainer).
+     */
+    func makeReleaseBody(
+        mergedBy: Components.Schemas.nullable_simple_user
+    ) async throws -> String {
+        let codeOwners = try await getCodeOwners()
+
+        let acknowledgment: String
+
+        let author = pr.user.login
+        let merger = mergedBy.login
+
+        /// If author is a code owner, skip crediting them as the author.
+        if codeOwners.contains(author) {
+            acknowledgment = "This patch was released by @\(author)."
+        } else {
+            if author == merger {
+                acknowledgment = "This patch was authored and released by @\(author)."
+            } else {
+                acknowledgment = "This patch was authored by @\(author) and released by @\(merger)."
+            }
+        }
+
+        return """
+        ###### _\(acknowledgment)_
+
+        "\(pr.body ?? "Pull Request:") \(pr.html_url)"
+        """
+    }
+
+    /// Returns code owners if the repo contains the file or returns `nil`.
+    /// In form of `["gwynne", "0xTim"]`.
+    func getCodeOwners() async throws -> Set<String> {
+        let fullName = repo.full_name.addingPercentEncoding(
+            withAllowedCharacters: .urlPathAllowed
+        ) ?? repo.full_name
+        let url = "https://raw.githubusercontent.com/\(fullName)/main/.github/CODEOWNERS"
+        let request = HTTPClientRequest(url: url)
+        let response = try await context.httpClient.execute(request, timeout: .seconds(3))
+        guard response.status == .ok else {
+            context.logger.debug("Can't find code owners of repo")
+            return []
+        }
+        let body = try await response.body.collect(upTo: 1 << 16)
+        let text = String(decoding: Data(buffer: body), as: UTF8.self)
+        let codeOwners = text.split(
+            omittingEmptySubsequences: true,
+            whereSeparator: \.isNewline
+        ).map {
+            $0.split(omittingEmptySubsequences: true, whereSeparator: \.isWhitespace)
+        }.flatMap {
+            $0.dropFirst().map {
+                String($0.dropFirst())
+            }
+        }
```

Comment:
This does not fully parse the `CODEOWNERS` format (see https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-syntax).
```suggestion
        let codeOwners = text
            // split into lines
            .split(omittingEmptySubsequences: true, whereSeparator: \.isNewline)
            // trim leading whitespace per line
            .map { $0.trimmingPrefix(\.isWhitespace) }
            // remove whole-line comments
            .filter { !$0.starts(with: "#") }
            // remove partial-line comments
            .compactMap { $0.split(separator: "#", maxSplits: 1, omittingEmptySubsequences: true).first } 
            // split lines on whitespace, dropping first word, and combine to single list
            .flatMap { $0.split(omittingEmptySubsequences: true, whereSeparator: \.isWhitespace).dropFirst() }
```
(Written in Discord, not tested, might be horribly horribly wrong 🤷‍♀️)